packages/eslint-plugin/lib/config/workleap.ts

```diff
@@ -0,0 +1,18 @@
+import type { Linter } from "eslint";
+import StrictCSSModuleNames from "../rules/strict-css-modules-names.ts";
+
+const config: Linter.FlatConfig = {
+    plugins: {
```

Comment:
If the idea if to import configs as JS objects or array and merge them on the consumer side, I don't get this plugin thing, what does it do?

Reply:
In ESLint plugins that have been properly updated, this `plugins` key is just part of the recommended config, and you don't have to manually provide it. Ideally, once everything is updated, you don't need to manually provide it unless you need to override something.

The other reason to provide it is to change the prefix of the rules. For example, typescript-eslint recommends manually providing the plugin to the "@typescript-eslint" key, so that any rules that start with `@typescript-eslint/` will be matched with those rules. typescript-eslint changed their package name so it isn't but they didn't want everyone to have to change all the rules, so they recommend this for backwards compatibility.

This is the same reason I provide the plugin directly for the react-hooks plugin, even though the rest of the plugin format is already technically compatible with flat config.

Long story short:

1. The purpose of the `plugin` key is to provide rules definitions to ESLint
2. Ideally, this would be part of the recommended config and you wouldn't have to manually provide it
3. There are still legitimate reasons to provide it manually, usually if the plugin's docs tell you to